tests/unit/test_logging.py

```diff
@@ -0,0 +1,224 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Tests for the log module."""
+
+import json
+import logging
+from contextlib import contextmanager
+from dataclasses import dataclass, field
+
+import pytest
+
+from hexkit.log import (
+    JsonFormatter,
+    LoggingConfig,
+    RecordCompiler,
+    configure_logging,
+)
+
+VALID_SERVICE_NAME = "test"
+VALID_INSTANCE_ID = "001"
+VALID_CONFIG = LoggingConfig(
+    log_level="INFO",
+    service_name="hexkit-test",
+    service_instance_id="0",
+)
+
+DEFAULT_KEYS = (  # all the keys present in the resulting log
+    "timestamp",
+    "service",
+    "instance",
+    "level",
+    "name",
+    "correlation_id",
+    "message",
+    "details",
+)
+
+ADDED_KEYS = (  # a list of non-standard keys added by the adapter/record compiler
+    "timestamp",
+    "service",
+    "instance",
+    "level",
+    "correlation_id",
+    "details",
+)
+
+DEFAULT_CONFIG = LoggingConfig(service_name="", service_instance_id="")
+
+
+@dataclass
+class JsonLog:
+    """A class to represent the Json Log format for convenience"""
+
+    timestamp: str = ""
+    service: str = ""
+    instance: str = ""
+    level: str = ""
+    name: str = ""
+    correlation_id: str = ""
+    message: str = ""
+    details: dict[str, str] = field(default_factory=dict)
+
+    def update_values_from_dict(self, values: dict[str, str]):
+        """Modify dataclass in place"""
+        self.__dict__.update(values)
+
+
+@pytest.fixture
+def expect_json_log(capsys):
+    """Fixture used to read logs from stderr and parse them as JSON for examination.
+
+    Raises:
+        - RuntimeError: when the log isn't in JSON format.
+    """
+
+    @contextmanager
+    def capture_log():
+        """Context manager to clear the bugger, capture input, and construct a JsonLog."""
+        capsys.readouterr()  # Clear the capture buffer
+        loaded_log: JsonLog = JsonLog()
+        yield loaded_log
+        captured = capsys.readouterr().err  # Capture the stderr
+        try:
+            loaded_log.update_values_from_dict(json.loads(captured))
+        except json.JSONDecodeError as err:
+            raise RuntimeError("Log doesn't seem to be in JSON format") from err
+
+    return capture_log
+
+
+def test_configured_log_level(caplog, expect_json_log):
+    """Test with config"""
+    logger = logging.getLogger("test_configured_log_level")
+    config = LoggingConfig(
+        log_level="CRITICAL",  # set to higher than default of WARNING
+        service_name=VALID_SERVICE_NAME,
+        service_instance_id=VALID_INSTANCE_ID,
+        log_format="",
+    )
+
+    configure_logging(logger=logger, config=config)
+
+    assert not caplog.records
```

Comment:
I thought the caplog fixture was not working propperly?

Reply:
It intercepts logs before formatting. It's fine for checking if a log has been emitted

Reply:
Ah OK, I see